crates/project_model/src/lib.rs

```diff
@@ -284,8 +313,10 @@ impl ProjectWorkspace {
     pub fn n_packages(&self) -> usize {
         match self {
             ProjectWorkspace::Json { project, .. } => project.n_crates(),
-            ProjectWorkspace::Cargo { cargo, sysroot } => {
-                cargo.packages().len() + sysroot.crates().len()
+            ProjectWorkspace::Cargo { cargo, sysroot, rustc } => {
+                let rustc_package_len = rustc.as_ref().map(|rc| rc.packages().len()).unwrap_or(0);
```

Comment:
```suggestion
                let rustc_package_len = rustc.as_ref().map_or(0, |rc| rc.packages().len());
```

Reply:
oh VERY cool, while i was writing that I wished there was a `map_or`...